### PR TITLE
feat/docs: Adding browser history documentation draft

### PR DESCRIPTION
**Pull request by MatthewHawkins**

### CODE REVIEW

File: docs/advanced/browser-history.md [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+## History size
+
+Understanding the size of the browser’s history stack can be useful when building features that depend on the user's navigation path. For example, you may want to <!-- vale off -->disable <!-- vale on -->certain navigation actions if there are no more entries to navigate to, or display a breadcrumb trail based on the history size.
+
+The `size()` method retrieves the total number of entries in the history stack.
+
+```java
+int historySize = history.size();
+System.out.println("History Length: " + historySize);
+```
```
Comment (hyyan) [inline docs/advanced/browser-history.md:113]: Let's remove this section

Comment (hyyan) [inline docs/advanced/browser-history.md:113]: Let's remove this paragraph

File: docs/advanced/browser-history.md [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+history.back();
+```
+
+- `forward()`: Moves the browser history forward by one step, simulating a user pressing the forward button in their browser. This works only if there’s an entry ahead in the history stack.
+
+  ```java
+  history.forward();
+  ```
+
+- `go(int index)`: Navigates to a specific point in the history stack based on an index. A positive number moves forward, a negative number moves backward, and zero reloads the current page. This method provides more granular control compared to back() and forward().
```
[the remark this answers is not in the document]
Comment (hyyan) [reply docs/advanced/browser-history.md:25]: back() and forward() needs to be wrapped with backticks

File: docs/advanced/browser-history.md (hunk 1 above)
Comment (hyyan) [inline docs/advanced/browser-history.md:113]: mention https://javadoc.io/doc/com.webforj/webforj-foundation/latest/com/webforj/Request.html

**Review by hyyan — CHANGES_REQUESTED** [review]

**Comment by hyyan** [comment]
@MatthewHawkins, it looks good overall. There's just one thing missing: explaining that you either initialize a new History object, or, if you're using routing, you can retrieve the created history instance from the router.

**Comment by hyyan** [comment]
@MatthewHawkins also we need to address the issues related to value.

**Comment by MatthewHawkins** [comment]
@hyyan - Made some changes, my first question is what you'd like me to mention specifically about the Request class (just to ensure I touch on all relevant points) https://javadoc.io/doc/com.webforj/webforj-foundation/latest/com/webforj/Request.html

Second thing would be discussing a couple vale rules I encountered here when we have time.

**Review by hyyan — APPROVED** [review]

**Comment by hyyan** [comment]
@MatthewHawkins this is good to go

**Pull request merged**